google/cloud/ndb/_datastore_api.py

```diff
@@ -146,8 +146,15 @@ def lookup(key, options):
                 entity_pb.MergeFromString(result)
 
             elif use_datastore:
-                yield _cache.global_lock(cache_key, read=True)
-                yield _cache.global_watch(cache_key)
+                lock_acquired = yield _cache.global_lock(cache_key, read=True)
```

Comment:
Transient errors that can't resolve and result in an exception should probably just be caught here and key_locked set to True as well, no need to bubble them up and fail things

Reply:
In writing this comment I just remembered that `handle_transient_errors` doesn't raise for read operations, which is the source of the problem described in #652, but which when addressed would then want a try/except here as described above.

Reply:
ebb2a1c

I believe this also addresses #652

Reply:
Agreed because it will return None, however I do think a try/catch is appropriate for non-transient errors.  Just because something fails with the cache even catastrophically should not prevent a read thread from going to the db for the value, it should just prevent it from writing back to the cache.